lib/acts_as_list/active_record/acts/list.rb

```diff
@@ -455,24 +455,31 @@ def update_positions
 
           def internal_scope_changed?
             return @scope_changed if defined?(@scope_changed)
-            
+
             @scope_changed = scope_changed?
           end
 
           # Temporarily swap changes attributes with current attributes
-          def swap_changed_attributes
-            @changed_attributes.each do |k, _|
-              if self.class.column_names.include? k
-                @changed_attributes[k], self[k] = self[k], @changed_attributes[k]
-              end
+          def with_old_attributes
+            begin
+              changed_attributes.each { |k, _| @changed_attributes[k], self[k] = self[k], @changed_attributes[k] }
```

Comment:
Just a thought to make this completely use the external interface. Could you not just do:

```ruby
cached_changes = changes

cached_changes.each { |attribute, values| self[attribute] = values[0] }
yield
cached_changes.each { |attribute, values| self[attribute] = values[1] }
```
Then we no longer have to worry about that instance variable.